lib/keystore.cc

```diff
@@ -96,6 +100,50 @@ static rpmRC write_key_to_disk(rpmPubkey key, string & dir, string & filename, i
     return rc;
 }
 
+rpmRC rebuild_dir(rpmtxn txn, keystore * store, rpmKeyring keys)
+{
+
+    auto macros = rpm::macros();
+    auto [mrc, keyringpath] = macros.expand("%{_keyringpath}");
+    while ('/' == keyringpath.back())
+	keyringpath.pop_back();
+    std::string path = rpm::expand_path({rpmtxnRootDir(txn), keyringpath});
+    std::string tmppath = keyringpath + ".tmp/";
+    std::string oldpath = keyringpath + ".rpmold/";
+    std::error_code ec = {};
+    rpmRC rc = RPMRC_OK;
+
+    macros.push("_keyringpath", "", tmppath.c_str(), 0);
+
+    rpmPubkey key = NULL;
+    auto iter = rpmKeyringInitIterator(keys, 0);
+    while ((rc == RPMRC_OK) && (key = rpmKeyringIteratorNext(iter))) {
+	rc = store->import_key(txn, key, 0, 0);
```

Comment:
This will stop on first error, which seems contrary to the goal of weeding out bad keys with warning.

Reply:
... Also this suggests that there shoudn't perhaps be a separate keystore-specific rebuild operation, ie it should just be handled by the rpmts-level rebuild - if we use the rpmdb backend as a model, that doesn't have a backend-specific rebuild operation either.